### ISSUE TITLE
[id-length] Extend ESLint core rule to also verify type names.

### ISSUE THREAD

**Issue by iliubinskii** [typescript-eslint/typescript-eslint#4779]
**Repro**

```TS
/*eslint id-length: ["error", { "max": 4 }]*/

const x1234567890 = 1;

type X1234567890 = string.
```

Shows warning for const, but not for type.

**Expected Result**

With "@typescript-eslint/id-length" should show warning for both.

**Comment by alexleonov-tactiq** [comment]
+1, should work for interface and type declarations

In the meantime, this can help in your ESLint configuration, if you can match the files:

```
"overrides": [
      {
        "files": "src/blablabla/types.ts",
        "rules": {
          "@typescript-eslint/naming-convention": [
            "error",
            { "selector": "property", "format": [], "custom": { "regex": "^.{1,25}$", "match": true } }
          ]
        }
      }
    ]
```

**Comment by Josh-Cena** [comment]
Is this actually worth pursuing when we have `naming-convention`? This feels like the same situation as `camelcase`.

**Comment by bradzacher** [comment]
yeah this has been open for close to 2 years and has only garnered 1 additional reaction.
additionally this core rule has existed longer than typescript-eslint, yet this is the first case of someone wanting this rule.

So ultimately either the rule itself is niche (I believe it is this because a lot of people like 1 letter variables in many cases) and / or drive to use it for types is low-to-non-existent and/or all its usecases are covered by naming-convention.

So definitely not something we'll be adding to our project.
Thanks for filing!
